dot/services_test.go

```diff
@@ -0,0 +1,656 @@
+		GenesisAuthorities: []types.AuthorityRaw{},
+		Randomness:         [32]byte{},
+		SecondarySlots:     0,
+	}
+	epochState, err := state.NewEpochStateFromGenesis(stateSrvc.DB(), stateSrvc.Block, genesisBABEConfig)
+	require.NoError(t, err)
+
+	stateSrvc.Epoch = epochState
+
+	rtCfg := &wasmer.Config{}
+
+	rtCfg.Storage, err = rtstorage.NewTrieState(genTrie)
+	require.NoError(t, err)
+
+	rtCfg.CodeHash, err = stateSrvc.Storage.LoadCodeHash(nil)
+	require.NoError(t, err)
+
+	rtCfg.NodeStorage = runtime.NodeStorage{}
+
+	rt, err := wasmer.NewRuntimeFromGenesis(rtCfg)
+	require.NoError(t, err)
+
+	stateSrvc.Block.StoreRuntime(stateSrvc.Block.BestBlockHash(), rt)
+
+	return stateSrvc
+}
+
+func Test_nodeBuilder_createCoreService(t *testing.T) {
+	t.Parallel()
+	cfg := NewTestConfig(t)
+	stateSrvc := newStateService(t)
+	ks := keystore.NewGlobalKeystore()
+	kr, err := keystore.NewSr25519Keyring()
+	require.NoError(t, err)
+	ks.Babe.Insert(kr.Alice())
+
+	networkService := &network.Service{}
+
+	type args struct {
+		cfg *Config
+		ks  *keystore.GlobalKeystore
+		st  *state.Service
+		net *network.Service
+		dh  *digest.Handler
+	}
+	tests := []struct {
+		name      string
+		args      args
+		expectNil bool
+		err       error
+	}{
+		{
+			name: "base case",
+			args: args{
+				cfg: cfg,
+				ks:  ks,
+				st:  stateSrvc,
+				net: networkService,
+			},
+			expectNil: false,
+			err:       nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			no := nodeBuilder{}
+			got, err := no.createCoreService(tt.args.cfg, tt.args.ks, tt.args.st, tt.args.net, tt.args.dh)
+
+			assert.ErrorIs(t, err, tt.err)
+
+			// TODO: create interface for core.NewService sa that we can assert.Equal the results
+			if tt.expectNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+				assert.IsType(t, &core.Service{}, got)
+			}
+		})
+	}
+}
+
+func Test_nodeBuilder_createNetworkService(t *testing.T) {
+	t.Parallel()
+	cfg := NewTestConfig(t)
+	stateSrvc := newStateService(t)
+	type args struct {
+		cfg             *Config
+		stateSrvc       *state.Service
+		telemetryMailer telemetry.Client
+	}
+	tests := []struct {
+		name      string
+		args      args
+		expectNil bool
+		err       error
+	}{
+		{
+			name: "base case",
+			args: args{
+				cfg:       cfg,
+				stateSrvc: stateSrvc,
+			},
+			expectNil: false,
+			err:       nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			no := nodeBuilder{}
+			got, err := no.createNetworkService(tt.args.cfg, tt.args.stateSrvc, tt.args.telemetryMailer)
+			assert.ErrorIs(t, err, tt.err)
+			// TODO: create interface for network.NewService to handle assert.Equal test
+			if tt.expectNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+				assert.IsType(t, &network.Service{}, got)
+			}
+		})
+	}
+}
+
+func Test_nodeBuilder_createRPCService(t *testing.T) {
+	t.Parallel()
+	cfg := NewTestConfig(t)
+	stateSrvc := newStateService(t)
```

Comment:
move this in subtest since it creates mocks

Reply:
Updated.